plugins/image-source/obs-slideshow-mk2.c

```diff
@@ -0,0 +1,1192 @@
+#include <obs-module.h>
+#include <util/circlebuf.h>
+#include <util/threading.h>
+#include <util/platform.h>
+#include <util/darray.h>
+#include <util/dstr.h>
+#include <util/task.h>
+
+#include <inttypes.h>
+
+#define do_log(level, format, ...)               \
+	blog(level, "[slideshow: '%s'] " format, \
+	     obs_source_get_name(ss->source), ##__VA_ARGS__)
+
+#define warn(format, ...) do_log(LOG_WARNING, format, ##__VA_ARGS__)
+
+/* clang-format off */
+
+#define S_TR_SPEED                     "transition_speed"
```

Comment:
In general we advise against the use of preprocessor defines for constants and use actual language constructs (e.g. `static const char *S_TR_SPEED = "transition_speed";` in this case) because they are type safe and play better with debuggers.

Reply:
I was able to do this for the settings values, but doing this for the translation defines would likely make the file much less clean because we'd have to use obs_module_text() everywhere rather than the macro. So in the case of translations, the macro has the specific purpose of trying to make the code cleaner.

I'd personally prefer keeping the macros for the translation values, but if you'd prefer me to just place obs_module_text() everywhere and convert these to static constants then that's no biggie either. Let me know which you'd prefer to see.

Reply:
I guess for function aliasing that's more a preference thing, so I'd be fine with it either way.